plugin/src/main/kotlin/org/jetbrains/research/pynose/plugin/util/AbstractTestSmellInspection.kt

```diff
@@ -0,0 +1,68 @@
+package org.jetbrains.research.pynose.plugin.util
+
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementVisitor
+import com.jetbrains.python.inspections.PyInspection
+import com.jetbrains.python.psi.PyClass
+import com.jetbrains.python.psi.PyFunction
+import org.jetbrains.research.pynose.core.PyNoseUtils
+import java.lang.reflect.InvocationTargetException
+import java.util.*
+import java.util.stream.Collectors
+
+abstract class AbstractTestSmellInspection : PyInspection() {
+
+    protected var testCase: PyClass? = null
+    protected var currentMethod: PyFunction? = null
+
+    abstract override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor
+
+    companion object {
+        private val LOG = Logger.getInstance(AbstractTestSmellInspection::class.java)
+
+        abstract class MyPsiElementVisitor() : PsiElementVisitor() {
+            override fun visitElement(element: PsiElement) {
```

Comment:
Changed it to a default PyElementVisitor. AbstractTestSmellInspection is not used anymore, however I decided not to remove it in case I need it to implement other inspections (or refer to it in the process).